doc/nrf/libraries/others/identity_key.rst

```diff
@@ -0,0 +1,38 @@
+.. _lib_identity_key:
+
+Identity key
+############
+
+.. contents::
+   :local:
+   :depth: 2
+
+The identity key library manages an asymmetric key used for identity services on devices with the Arm CryptoCell and KMU peripherals.
+
+Functionality
+*************
+
+This library manages the identity keys, which are asymmetric keys intended to provide identity to specific devices.
+The identity key is designed to be unique and is either device-generated or provisioned in a secure manner during production.
```

Comment:
I see we have two ways to create the identity key, but I only understand the use-case for one of them.

According to the "PSA Security Model":

"At manufacture, the device is provisioned with immutable PSA parameters, see Table 2, that are
sufficient to uniquely and securely identify the device instance." --  https://developer.arm.com/-/media/Files/pdf/PlatformSecurityArchitecture/Architect/DEN0079_PSA_SM_ALPHA-03_RC01.pdf

So the use-case for provisioning a unique key is clear.

But when would one want to generate a unique key on the device? Is this for non-PSA use-cases?

Reply:
This either-or sentence is implying that you either device-generate a key, or provision in a secure manner.

But you can provision in a secure manner by device-generating a key.

This sentence needs to be re-written to remove the implication that it's either or.

Reply:
Rewrote the line, fixed